src/renderers/webxr/WebXRManager.js

```diff
@@ -348,6 +362,9 @@ class WebXRManager extends EventDispatcher {
 				customReferenceSpace = null;
 				referenceSpace = await session.requestReferenceSpace( referenceSpaceType );
 
+				currentPixelRatio = renderer.getPixelRatio();
+				renderer.getDrawingBufferSize( currentSize );
```

Comment:
I think you have to use `getSize()` here. Otherwise the renderer gets a too large size when later using `setDrawingBufferSize()`. 

The problem is that the vector returned from `getDrawingBufferSize()` is the effective resolution meaning dimension * pixel ratio. When restoring the renderer's size via `setDrawingBufferSize()`, you have to apply width and height without the pixel ratio honored (because the pixel ratio is applied as the third parameter).

Reply:
If the drawing buffer methods are too confusing from your point of view, consider to use a combination of `setSize()`/`setPixelRatio()` and `getSize()`/`getPixelRatio()`. Maybe the code gets easier to read with this.